### PR TITLE
Support kubeflow reuse mode

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: nni/experiment/config/kubeflow.py [hunk 1]
```diff
@@ -8,21 +8,21 @@
...
     azure_account: Optional[str] = None
     azure_share: Optional[str] = None
-    key_vault: Optional[str] = None
-    key_vault_secret: Optional[str] = None
+    key_vault_name: Optional[str] = None
+    key_vault_key: Optional[str] = None
 
 @dataclass(init=False)
-class KubeflowNfsConfig(_KubeflowStorageConfig):
+class KubeflowNfsConfig(KubeflowStorageConfig):
```
Comment (liuzhe-lz) [inline nni/experiment/config/kubeflow.py:25]: Shouldn't reusable use shared storage service?

File: nni/experiment/config/kubeflow.py [hunk 2]
```diff
@@ -32,37 +32,41 @@ class KubeflowAzureStorageConfig(ConfigBase):
...
+    master: Optional[KubeflowRoleConfig] = None
+    reuse_mode: Optional[bool] = True #set reuse mode as true for v2 config
 
     def __init__(self, **kwargs):
         kwargs = util.case_insensitive(kwargs)
-        kwargs['storage'] = util.load_config(_KubeflowStorageConfig, kwargs.get('storage'))
+        kwargs['storage'] = util.load_config(KubeflowStorageConfig, kwargs.get('storage'))
         kwargs['worker'] = util.load_config(KubeflowRoleConfig, kwargs.get('worker'))
-        kwargs['parameterserver'] = util.load_config(KubeflowRoleConfig, kwargs.get('parameterserver'))
+        kwargs['ps'] = util.load_config(KubeflowRoleConfig, kwargs.get('ps'))
```
Comment (J-shang) [inline nni/experiment/config/kubeflow.py:65]: `ps` is `parameterserver`? why we don't have `master` before?

**Review by ultmaster — APPROVED** [review]

File: nni/experiment/config/kubeflow.py (hunk 2 above)
Comment (liuzhe-lz) [reply nni/experiment/config/kubeflow.py:65]: I once merged ps and master. They are both parameter server. No difference at all, just different names for pytorch and tensorflow.

**Review by J-shang — APPROVED** [review]

File: nni/experiment/config/kubeflow.py (hunk 1 above)
Comment (SparkSnail) [reply nni/experiment/config/kubeflow.py:25]: Only openPAI environment service support storage service.

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**